sdk/core/azure-core/src/main/java/com/azure/core/credential/TokenRefreshOptions.java

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.credential;
+
+import java.time.Duration;
+
+/**
+ * The options to configure the token refresh behavior.
+ */
+public class TokenRefreshOptions {
+    private static final Duration DEFAULT_REFRESH_TIMEOUT = Duration.ofSeconds(30);
+    private static final Duration DEFAULT_REFRESH_OFFSET = Duration.ofMinutes(2);
+
+    private Duration tokenRefreshTimeout = DEFAULT_REFRESH_TIMEOUT;
+    private Duration tokenRefreshOffset = DEFAULT_REFRESH_OFFSET;
+
+    /**
+     * Returns a Duration value representing the amount of time to wait between token refreshes. This is to prevent
+     * sending too many requests to the authentication service.
+     *
+     * @return the duration value representing the amount of time to wait between token refreshes
+     */
+    public Duration getTokenRefreshTimeout() {
+        return tokenRefreshTimeout;
+    }
+
+    /**
+     * Specifies a Duration value representing the amount of time to wait between token refreshes. This is to prevent
+     * sending too many requests to the authentication service.
+     *
+     * @param tokenRefreshTimeout the duration value representing the amount of time to wait between token refreshes
+     * @return TokenRefreshOptions
+     */
+    public TokenRefreshOptions setTokenRefreshTimeout(Duration tokenRefreshTimeout) {
```

Comment:
should this be setTokenRefreshRetryTimeout ?

TokenRefreshTimeout can be confused with TokenRefreshOffset

Reply:
Updated.